### PR TITLE
Plugin documentation

### PR DESCRIPTION
**Pull request by soupette**
My PR is a:
- [ ] 💥 Breaking change
- [ ] 🐛 Bug fix
- [ ] 💅 Enhancement
- [x] 🚀 New feature

Main update on the:
- [x] Admin
- [ ] Documentation
- [ ] Framework
- [x] Plugin

API documentation generated with swagger.

### CODE REVIEW

File: packages/strapi-plugin-documentation/documentation/1.0.0/full_documentation.json [hunk 1]
```diff
@@ -0,0 +1,2468 @@
+{
```
Comment (soupette) [inline packages/strapi-plugin-documentation/documentation/1.0.0/full_documentation.json]: This file needs to de removed.

File: packages/strapi-plugin-email/documentation/1.0.0/email-Email.json [hunk 2]
```diff
@@ -0,0 +1,80 @@
+{
```
Comment (soupette) [inline packages/strapi-plugin-email/documentation/1.0.0/email-Email.json]: This file needs to be removed as well.

File: packages/strapi-plugin-upload/documentation/1.0.0/upload-File.json [hunk 3]
```diff
@@ -0,0 +1,396 @@
+{
```
Comment (soupette) [inline packages/strapi-plugin-upload/documentation/1.0.0/upload-File.json]: Same for this file

File: packages/strapi-plugin-users-permissions/documentation/1.0.0/users-permissions-Role.json [hunk 4]
```diff
@@ -0,0 +1,689 @@
+{
```
Comment (soupette) [inline packages/strapi-plugin-users-permissions/documentation/1.0.0/users-permissions-Role.json]: Same for this file

File: packages/strapi-plugin-users-permissions/documentation/1.0.0/users-permissions-User.json [hunk 5]
```diff
@@ -0,0 +1,1200 @@
+{
```
Comment (soupette) [inline packages/strapi-plugin-users-permissions/documentation/1.0.0/users-permissions-User.json]: Same for this file

File: packages/strapi/bin/strapi-start.js [hunk 6]
```diff
@@ -66,7 +66,7 @@ module.exports = function(appPath = '') {
       };
 
       const setFilesToWatch = (src) => {
-        let files = _.includes(src, '/admin') || _.includes(src, 'components') ? [] : fs.readdirSync(src);
+        let files = _.includes(src, '/admin') || _.includes(src, 'components') || _.includes(src, '/docmentation') ? [] : fs.readdirSync(src);
```
Comment (soupette) [inline packages/strapi/bin/strapi-start.js]: Typo: `_.includes(src, '/documentation') ?`

File: scripts/setup.js [hunk 7]
```diff
@@ -112,33 +112,26 @@ watcher('📦  Linking strapi-hook-bookshelf...', 'npm link');
...
 shell.cd('../strapi-provider-email-sendmail');
-watcher(
-  '📦  Linking strapi-provider-email-sendmail...',
-  'npm link --no-optional',
-  false
-);
+watcher('📦  Linking strapi-provider-email-sendmail...', 'npm link --no-optional', false);
 
 // Plugins with admin
+shell.cd('../strapi-plugin-documentation');
```
Comment (soupette) [inline scripts/setup.js]: We should create a function here

File: packages/strapi-plugin-users-permissions/documentation/1.0.0/overrides/users-permissions-User.json [hunk 8]
```diff
@@ -0,0 +1,281 @@
+{
+  "paths": {
+    "/auth/local/register": {
+      "post": {
```
Comment (derrickmehaffy) [inline packages/strapi-plugin-users-permissions/documentation/1.0.0/overrides/users-permissions-User.json]: This route should have an override on security as it doesn't require authentication to use.

Example to remove authentication within swagger:

```
  "paths": {
    "/auth/local/register": {
      "post": {
        "tags": ["Authentication"],
        "security": [],
        "externalDocs": {
          "description": "Find out more in the strapi's documentation",
```

Comment (derrickmehaffy) [reply packages/strapi-plugin-users-permissions/documentation/1.0.0/overrides/users-permissions-User.json]: image

File: packages/strapi-plugin-users-permissions/documentation/1.0.0/overrides/users-permissions-User.json [hunk 9]
```diff
@@ -0,0 +1,281 @@
...
+        },
+        "responses": {
+          "200": {
+            "description": "Your user is redirected"
+          }
+        }
+      }
+    },
+    "/auth/local": {
+      "post": {
```
Comment (derrickmehaffy) [inline packages/strapi-plugin-users-permissions/documentation/1.0.0/overrides/users-permissions-User.json]: Also shouldn't require authentication, see other example

**Comment by derrickmehaffy** [comment]
I'll work on creating another PR to document how users can modify the security requirements in overrides on each model. For now it would be far too difficult to query the database (especially if the project hasn't been started yet IE the database isn't created) and since permission changes don't require server restarts anymore it would really hurt in production.

**Comment by derrickmehaffy** [comment]
PR created for my working changes: https://github.com/strapi/strapi/pull/2462

File: packages/strapi-plugin-users-permissions/documentation/1.0.0/overrides/users-permissions-User.json (hunk 8 above)
Comment (derrickmehaffy) [reply packages/strapi-plugin-users-permissions/documentation/1.0.0/overrides/users-permissions-User.json]: Fixed with merged PR https://github.com/strapi/strapi/pull/2462

File: packages/strapi-plugin-users-permissions/documentation/1.0.0/overrides/users-permissions-User.json (hunk 9 above)
Comment (derrickmehaffy) [reply packages/strapi-plugin-users-permissions/documentation/1.0.0/overrides/users-permissions-User.json]: Fixed with merged PR https://github.com/strapi/strapi/pull/2462

**Comment by soupette** [comment]
Thanks @derrickmehaffy

File: packages/strapi-plugin-documentation/services/Documentation.js [hunk 10]
```diff
@@ -0,0 +1,1616 @@
...
+      fs.accessSync(this.getMergedDocumentationPath());
+      return true;
+    } catch (err) {
+      return false;
+    }
+  },
+
+  /**
+   * Recursively create missing directories
+   * @param {String} apiName
```
Comment (lauriejim) [inline packages/strapi-plugin-documentation/services/Documentation.js]: Param name is `targetDir `

**Comment by lauriejim** [comment]
It works pretty well @soupette !! Waiting some update about function params.

**Comment by lauriejim** [comment]
Thank you @derrickmehaffy for this feedback.
We will fix it in a future patch.

**Review by tang88888888 — APPROVED** [review]

**Pull request merged**